pandas/core/array_algos/take.py

```diff
@@ -82,8 +82,13 @@ def _take_nd_ndarray(
     allow_fill: bool,
 ) -> np.ndarray:
 
+    if indexer is None:
+        indexer = np.arange(arr.shape[axis], dtype=np.int64)
+        dtype, fill_value = arr.dtype, arr.dtype.type()
+    else:
+        indexer = ensure_int64(indexer, copy=False)
```

Comment:
This part is copied from `_take_preprocess_indexer_and_fill_value`, so we can avoid doing that in `take_1d`